test/test_datapipe.py

```diff
@@ -569,6 +570,27 @@ def _return_self(x):
         # __len__ Test: returns the number of batches
         self.assertEqual(3, len(batch_dp))
 
+    def test_flatmap_datapipe(self):
+        source_dp = IterableWrapper(list(range(20)))
+
+        def fn(e):
+            return [e, e*10]
+
+        flatmapped_dp = nested_wrapper.flatmap(fn)
+        list_flatmapped_dp = list(flatmapped_dp)
+        expected_list = list(itertools.chain([(e, e*10) for e in source_dp]))
```

Comment:
Should we do another `self.assertEqual(list_flatmapped_dp, expected_list)` here?